zebra-state/src/service/non_finalized_state/chain.rs

```diff
@@ -737,6 +737,16 @@ impl Chain {
             .collect()
     }
 
+    /// Returns the Sapling note commitment subtree of the tip of this [`Chain`],
+    pub fn sapling_subtree_for_tip(&self) -> Option<NoteCommitmentSubtree<sapling::tree::Node>> {
+        if !self.is_empty() {
+            let tip = self.non_finalized_tip_height();
+            Some(self.sapling_subtree(tip.into())?)
+        } else {
+            None
+        }
```

Comment:
Optional/Nitpick:

```suggestion
        let (&index, data) = self.sapling_subtrees.last_key_value()?;
        Some(data.with_index(index))
```

Reply:
This suggestion changes the behaviour of the function from "completed at tip" to "most recently completed". I'm not sure that's what we want:
https://github.com/ZcashFoundation/zebra/pull/7636/files#r1340708466

Reply:
I think we should not apply this suggestion and return the subtree that was completed at tip (current behaviour), which is the same we did for `ZebraDb`.